pkg/receive/handler.go

```diff
@@ -747,34 +755,55 @@ func (h *Handler) fanoutForward(pctx context.Context, tenant string, wreqs map[e
 	// This is needed if context is canceled or if we reached success of fail quorum faster.
 	defer func() {
 		go func() {
-			for err := range ec {
+			for errReplicaFn := range ec {
+				_, err := errReplicaFn()
 				if err != nil {
 					level.Debug(tLogger).Log("msg", "request failed, but not needed to achieve quorum", "err", err)
 				}
 			}
 		}()
 	}()
 
-	var success int
+	var (
+		// success counts how many requests per replica have succedeed.
```

Comment:
Could we reflect this comment and all the others that follow into their respective variable name? This pattern of comment explaining "what" and not "why" is often a sign that something deserves a better name.

Reply:
I tried to name it a bit, although I'm finding it challenging to try and find clear enough name that it would encapsulate the complexity of replication logic and be obvious without additional commentary. Suggestions welcome.